### ISSUE TITLE
Looped video randomly stops

### ISSUE THREAD

**Issue by itlancer** [Gamua/Adobe-Runtime-Support#79]
## Problem Description
Looped video randomly stops. It could take about 2 minutes short video looped playback to see what happens. Sometimes it stops just after first time playback. When it stops it show last frame or white/black rectangle instead of video or flickering. But by Scout you can see that something happends with video decoding, frame processing after it stops.
Tested with multiple devices across many OSes: Windows 8.1, Windows 10, Android 6/7/8/9. With iOS it doesn't work at all because of this: https://tracker.adobe.com/#/view/AIR-3840986
It has been tested with many different AIR versions and devices and its architectures even with latests AIR 32.0.0.144 beta and AIR 33.0.1.220.
Same problem with all cases. It works fine only with macOS.
Changing renderMode in application manifest doesn't help. Changing containsVideo in application manifest for Android doesn't help.  `NetStream::useHardwareDecoder = true` with Windows doesn't help.

## Steps to Reproduce
Launch code below. Better to use short video 1-3 seconds duration.
Application example with sources and example of video attached.
[video_loop_stuck.zip](https://github.com/Gamua/Adobe-Runtime-Support/files/3520085/video_loop_stuck.zip)

```
package  {
	import flash.display.Sprite;
	import flash.events.Event;
	import flash.net.NetConnection;
	import flash.net.NetStream;
	import flash.media.Video;
	import flash.events.NetStatusEvent;
	import flash.desktop.NativeApplication;
	import flash.desktop.SystemIdleMode;
	
	public class VideoLoopStuck extends Sprite {
		private var nc:NetConnection;
		private var ns:NetStream;
		private var video:Video = new Video(640, 480);
		
		public function VideoLoopStuck() {
			addEventListener(Event.ADDED_TO_STAGE, init);
		}
		
		private function init(e:Event):void {
			removeEventListener(Event.ADDED_TO_STAGE, init);
			
			NativeApplication.nativeApplication.systemIdleMode = SystemIdleMode.KEEP_AWAKE;
			addChild(video);
			
			nc = new NetConnection();
			nc.addEventListener(NetStatusEvent.NET_STATUS, ncHandler);
			nc.connect(null);
		}
		
		private function ncHandler(e:NetStatusEvent):void {
			if (e.info.code == "NetConnection.Connect.Success"){
				ns = new NetStream(nc);
				ns.client = {onMetaData:getMeta};
				ns.addEventListener(NetStatusEvent.NET_STATUS, nsHandler);
				video.attachNetStream(ns);
				ns.play("video.mp4");
			}
		}

		private function nsHandler(e:NetStatusEvent):void {
			trace(e.info.code);
			if (e.info.code == "NetStream.Play.Stop"){
				ns.seek(0);
			}
		}

		private function getMeta(mdata:Object):void { }
		
	}
}
```

**Actual Result:**
After few minutes (may be after first time playback) video stops. When it stops it show last frame or white/black rectangle instead of video or flickering.

**Expected Result:**
Video playback in loop smoothly.

## Known Workarounds
none 
*- restart playback by NetStream::play() method as a new one file
*- [Android] change disableMediaCodec in application manifest to true, but it cause some performance problems and some videos doesn't play at all

**Comment by ajwfrost** [comment]
Hi @itlancer  - are you able to check this with the latest release? There were some changes made to fix this and the above test case is working for us now.
(Just on Android but the same code will be used to build the other platforms next..)

**Comment by itlancer** [comment]
Hi, @ajwfrost 
Cannot reproduce any more this issue with latest 33.0.2.330 with Android devices. Seems it fixed.
Waiting for other platforms in next releases.
Thank you!

**Comment by itlancer** [comment]
Seems it also fixed with AIR 33.0.2.338 for Windows. Waiting fixes for macOS and iOS.

**Comment by itlancer** [comment]
This issue fixed with latests AIR 33.0.2.338 and 33.1.0.16 beta.
Cannot be tested with iOS and tvOS because of https://github.com/Gamua/Adobe-Runtime-Support/issues/211
Also there is some other issues that influence to correct video playback across different platforms:
https://github.com/Gamua/Adobe-Runtime-Support/issues/80
https://github.com/Gamua/Adobe-Runtime-Support/issues/155
https://github.com/Gamua/Adobe-Runtime-Support/issues/151
https://github.com/Gamua/Adobe-Runtime-Support/issues/139
https://github.com/Gamua/Adobe-Runtime-Support/issues/100
https://github.com/Gamua/Adobe-Runtime-Support/issues/98
https://github.com/Gamua/Adobe-Runtime-Support/issues/93
https://github.com/Gamua/Adobe-Runtime-Support/issues/92
https://github.com/Gamua/Adobe-Runtime-Support/issues/89
https://github.com/Gamua/Adobe-Runtime-Support/issues/88
https://github.com/Gamua/Adobe-Runtime-Support/issues/87
https://github.com/Gamua/Adobe-Runtime-Support/issues/82
https://github.com/Gamua/Adobe-Runtime-Support/issues/81
https://github.com/Gamua/Adobe-Runtime-Support/issues/69
https://github.com/Gamua/Adobe-Runtime-Support/issues/15
https://github.com/Gamua/Adobe-Runtime-Support/issues/9

and feature requests:
https://github.com/Gamua/Adobe-Runtime-Support/issues/180
https://github.com/Gamua/Adobe-Runtime-Support/issues/197
https://github.com/Gamua/Adobe-Runtime-Support/issues/171
https://github.com/Gamua/Adobe-Runtime-Support/issues/141
https://github.com/Gamua/Adobe-Runtime-Support/issues/68
